### PR TITLE
#160090098 Fix  asset category update issue

### PR DESCRIPTION
**Pull request by appcypher**
#### What does this PR do?
- This task fixes a bug that causes asset category custom attributes not to change when updated via the `/api/v1/asset-categories`

    **Current Behavior**
    When the custom attributes of an asset category is updated via the `/api/v1/asset-categories` endpoint, the custom attributes don't get updated in the database.

    **Expected Behavior**
    An asset category's custom attributes should be successfully updated in the database when a PATCH request is sent to `/api/v1/asset-categories` endpoint.

#### Description of Task to be completed?
- Fix the issue causing custom attributes changes not reflecting in the database

#### How should this be manually tested?
* Fetch the branch: `git fetch bg-fix-asset-category-update-issue-160090098`
* Checkout to the branch: `git checkout bg-fix-asset-category-update-issue-160090098`
* Switch to the app environment: `pipenv shell` (you need `zsh`, `bash` or `sh` for this)
* Seed the database: `flask seed_database`
* Run the application: `flask run`
* Open `postman` and try sending a PATCH request to `/api/v1/asset-categories`
* You will see the updated changes in the response body
* **Sample Request Body**
    ```json
    {
        "name": "Apple Hello",
        "customAttributes": [
            {
                "id": "-LL-4Vza6DqSwo2DF-9o",
                "inputControl": "text",
                "isRequired": false,
                "key": "warranty",
                "label": "Warranty"
            }
        ]
    }
    ```

#### Any background context you want to provide?
- N/A

#### What are the relevant pivotal tracker stories?
[#160066284](https://www.pivotaltracker.com/story/show/160066284)

#### Postman collection:
- N/A

#### Screenshot
- N/A

### CODE REVIEW

**Comment by rogerokello** [comment]
@appcypher Great work. Just some minor corrections in your PR description, under: 
1. `What does this PR do?` on the section `Current Behaviour`, I think the spelling should be endpoint not enpoint.
 2. `How should this be manually tested`. Are there some instructions missing. For example, git checkout develop, then git pull. Because I've tested your branch locally and it is working only I had to do checkout to develop and do a git pull to pull your branch.

Also, a quick question, does the command flask seed_database come before flask run?

**Comment by appcypher** [comment]
1. Thanks fixed 🎉 
2. That's weird. You don't need to pull develop for this branch to work, because it was branched off develop

File: api/utilities/model_serializers/attribute.py [hunk 1]
```diff
@@ -19,6 +19,7 @@ class AttributeSchema(BaseSchema):
     """Attribute model schema"""
 
     _key = fields.String(load_from='label', dump_to='key')
```
Comment (SEUNAGBEYE) [inline api/utilities/model_serializers/attribute.py]: @appcypher please make this `load_only`

Comment (SEUNAGBEYE) [reply api/utilities/model_serializers/attribute.py]: @appcypher This is yet to be fixed.

Comment (appcypher) [reply api/utilities/model_serializers/attribute.py]: Fixed! 🎉

**Comment by appcypher** [comment]
Fixed :tada:

File: tests/mocks/asset_category.py [hunk 2]
```diff
@@ -1,6 +1,21 @@
...
+        "inputControl": "text",
+    }, {
+        "label": "color",
+        "isRequired": True,
+        "inputControl": "dropdown",
+        "choices": ["blue", "red", "black"]
+    }]
+}
+
+valid_asset_category_data_with_keys = {  # pylint: disable=C0103
```
Comment (SEUNAGBEYE) [inline tests/mocks/asset_category.py]: could you please rename this to `valid_asset_category_data_with_attr_keys` to be explicit.

Comment (appcypher) [reply tests/mocks/asset_category.py]: Makes sense. Fixed 🎉

**Pull request merged**